web_client/analysis-modules/feature-matching.jsx

```diff
@@ -0,0 +1,170 @@
+
+import actions from '../actions';
+import analysisUtils from '../utils/analysis';
+import { Promise } from '../utils/promise';
+import { store, rest } from '../globals';
+
+import { formatSize } from 'girder/misc';
+
+const D = store.dispatch.bind(store);
+
+let page1;
+let computeResultElement;
+let applyResultElements;
+let tabGroupElement;
+let computeTabElement;
+let applyTabElement;
+let matchChooserElement;
+
+const computeMatch = (data, page) => {
+  const truncatePromise = Promise.all([
```

Comment:
I should have been more clear with my coined phrase "list construction" - I just meant the style you were using to highlight elements of the array you were constructing.

The main way I would improve this is to use a `const` variable for each of the promises, then deliver them all together afterwards with a simpler `Promise.all()` expression:

```javascript
const fooPromise = ...;
const barPromise = ...;
const bazPromise = ...;

const truncatePromise = Promise.all([fooPromise, barPromise, bazPromise]);
```

The drawback is that this declares more variables, but they are declared `const` so the potential damage from doing so is bounded. I think the gain in readability justifies the extra declarations. But again, this is a style thing so rely on your own preferences. As in other discussions, I'll just reiterate that the extra parentheses and linebreaks may generally impede readability because they're not constructions found in typical JS code.